### PR TITLE
[release/1.4 backport] selinux: relabel /dev/shm

### PR DESCRIPTION
**Pull request by dweomer**
Address an issue originally seen in the k3s 1.3 and 1.4 forks of containerd/cri, https://github.com/rancher/k3s/issues/2240.

This is a backport of containerd/containerd#4699

Even with updated container-selinux policy, container-local /dev/shm will get
mounted with container_runtime_tmpfs_t because it is a tmpfs created by the
runtime and not the container (thus, container_runtime_t transition rules apply).
The relabel mitigates such, allowing envoy proxy to work correctly (and other
programs that wish to write to their /dev/shm) under selinux.

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

File: pkg/server/container_create_unix.go [hunk 1]
```diff
@@ -97,13 +97,16 @@ func (c *criService) containerMounts(sandboxID string, config *runtime.Container
...
+			selinuxRelabel = false
 		}
 		mounts = append(mounts, &runtime.Mount{
-			ContainerPath: devShm,
-			HostPath:      sandboxDevShm,
-			Readonly:      false,
+			ContainerPath:  devShm,
+			HostPath:       sandboxDevShm,
+			Readonly:       false,
+			SelinuxRelabel: selinuxRelabel,
```
Comment (crosbymichael) [inline pkg/server/container_create_unix.go]: instead of a bool, you could just `SelinuxRelabel: sandboxDevShm != devShm,`

**Comment by dweomer** [comment]
Verified that this fixes an actual problem in our 1.3 and 1.4 forks for k3s.

**Review by estesp — APPROVED** [review]
LGTM

**Review by dmcgowan — APPROVED** [review]
LGTM

**Comment by dweomer** [comment]
@mikebrow, @dmcgowan these failures seem unrelated to my changes, unless the `pull-cri-containerd-node-e2e` hook is running on an selinux system?

**Pull request merged**